llvm/tools/sycl-post-link/CompileTimePropertiesPass.cpp

```diff
@@ -154,6 +241,36 @@ PreservedAnalyses CompileTimePropertiesPass::run(Module &M,
     }
   }
 
+  // Process all properties on kernels.
+  for (Function &F : M) {
+    // Only consider kernels.
```

Comment:
`device_has` will need to be handled differently as it is not meant to reach the backend but is instead needed in the information passed to the runtime. I won't rule out that it will be done around here, but neither do I feel confident enough in leaving a TODO here about it.